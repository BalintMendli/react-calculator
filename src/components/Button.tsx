import React from 'react';
import styled from 'styled-components/macro';
import { Label, Id, Pos } from 'constants/buttons';
import { Color } from 'constants/colors';
import { getGridPos } from 'logic/utils';

interface ButtonProps {
  label: Label;
  id: Id;
  pos: Pos;
  color: Color;
  handleClick: (label: Label) => void;
}

interface StyledProps {
  color: Color;
  gridPos: [string, string];
}

const StyledButton = styled.button<StyledProps>`
  grid-column: ${({ gridPos }): string => gridPos[0]};
  grid-row: ${({ gridPos }): string => gridPos[1]};
  background: ${({ color }): string => color};
  padding: 0;
  font: inherit;
  font-size: 1.1rem;
  outline: inherit;
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid #000;
  border-radius: 5px;
  cursor: pointer;
  user-select: none;
  opacity: 0.85;
  transition: opacity 250ms ease-in-out;

  :hover {
    opacity: 1;
  }
`;

const Button = ({
  label,
  id,
  handleClick,
  pos,
  color,
}: ButtonProps): JSX.Element => {
  const handleBtnClick = (): void => {
    handleClick(label);
  };

  return (
    <StyledButton
      type="button"
      id={id}
      color={color}
      gridPos={getGridPos(pos)}
      onClick={handleBtnClick}
    >
      {label}
    </StyledButton>
  );
};

export default Button;
