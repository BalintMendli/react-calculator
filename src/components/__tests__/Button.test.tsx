import React from 'react';
import Button from 'components/Button';
import { shallow, ShallowWrapper } from 'enzyme';

describe('BtnPanel component', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Button
        label="0"
        id="zero"
        color="#42464a"
        pos={{ x: 6, y: 1 }}
        handleClick={jest.fn()}
      />,
    );
  });

  it('renders correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders button div', () => {
    expect(wrapper.find('[type="button"]').length).toEqual(1);
    expect(wrapper.find('[type="button"]').text()).toEqual('0');
  });
});
