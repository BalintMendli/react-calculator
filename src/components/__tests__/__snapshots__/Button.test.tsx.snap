// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnPanel component renders correctly 1`] = `
"<Button__StyledButton type=\\"button\\" id=\\"zero\\" color=\\"#42464a\\" gridPos={{...}} onClick={[Function: handleBtnClick]}>
  0
</Button__StyledButton>"
`;
